mos_tests/neutron/python_tests/test_ovs_restart.py

```diff
@@ -0,0 +1,208 @@
+            ip_protocol='tcp',
+            from_port=1,
+            to_port=65535,
+            cidr='0.0.0.0/0')
+        self.os_conn.nova.security_group_rules.create(
+            default_sec_group.id,
+            ip_protocol='udp',
+            from_port=1,
+            to_port=65535,
+            cidr='0.0.0.0/0')
+
+    def disable_ovs_agents_on_controllers(self):
+        """Disable openvswitch-agents on all controllers."""
+        controllers = self.env.get_nodes_by_role('controller')
+        assert len(controllers) > 0, 'No controllers have been found.'
+
+        for node in controllers:
+            with self.env.get_ssh_to_node(node.data['ip']) as remote:
+                result = remote.execute(
+                    '. openrc && pcs resource enable '
+                    'p_neutron-plugin-openvswitch-agent --wait')
+                assert result['exit_code'] == 0
+
+    def restart_ovs_agents_on_computes(self):
+        """Restart openvswitch-agents on all computes."""
+        computes = self.env.get_nodes_by_role('compute')
+        assert len(computes) > 0, 'No computes have been found.'
+
+        for node in computes:
+            with self.env.get_ssh_to_node(node.data['ip']) as remote:
+                result = remote.execute(
+                    'service neutron-plugin-openvswitch-agent restart')
+                assert result['exit_code'] == 0
+
+    @contextmanager
+    def start_ping_from_vm(self, vm, vm_keypair, ip_to_ping=None):
+        """Start ping second vm from the first before enter and stop it after.
+        Log will be copied to /tmp/ovs_restart_ping.log path."""
+        def check_ping_from_vm(vm, vm_keypair, ip_to_ping):
+            if ip_to_ping is None:
+                ip_to_ping = settings.PUBLIC_TEST_IP
+
+            cmd = 'ping {0} > /tmp/ovs_restart_ping.log &'.format(ip_to_ping)
+
+            res = self.run_on_vm(vm, vm_keypair, cmd, vm_login="cirros",
+                                 timeout=3 * 60)
+            assert res['exit_code'] == 0, \
+                'Can\'t connect to vm .'.format(vm.name)
+
+        process = Process(target=check_ping_from_vm,
+                          args=(vm, vm_keypair, ip_to_ping))
+
+        try:
+            process.start()
+            yield
+        finally:
+            cmd = 'killall -sigint ping'
+            res = self.run_on_vm(vm, vm_keypair, cmd, vm_login="cirros",
+                                 timeout=3 * 60)
+            assert res['exit_code'] == 0, \
+                'Can\'t kill ping process with pid {}.'.format(res)
+
+            cmd = 'cat /tmp/ovs_restart_ping.log'
+            res = self.run_on_vm(vm, vm_keypair, cmd, vm_login="cirros",
+                                 timeout=3 * 60)
+            assert res['exit_code'] == 0, \
+                'Can\'t get log file /tmp/ovs_restart_ping.log ' \
+                'on vm {}.'.format(vm.name)
+
+            with open('/tmp/ovs_restart_ping.log', 'w') as log_file:
```

Comment:
In disable_ovs_agents_on_controllers method the 'pcs resource disable p_neutron-plugin-openvswitch-agent' operation, which goes before 'pcs resource enable p_neutron-plugin-openvswitch-agent', is absent.